python/ray/experimental/serve/metric.py

```diff
@@ -0,0 +1,154 @@
+import time
+
+import ray
+import numpy as np
+import pandas as pd
+
+
+@ray.remote
+class MetricMonitor:
+    def __init__(self, gc_window_seconds=3600):
+        """Metric monitor scrapes metrics from ray serve actors
+        and allow windowed query operations.
+
+        Args:
+            gc_window_seconds(int): How long will we keep the metric data in
+                memory. Data older than the gc_window will be deleted.
+        """
+        self.actor_handles = []
+
+        self.data_entries = []
+
+        self.gc_window_seconds = gc_window_seconds
+        self.latest_gc_time = time.time()
+
+    def add_target(self, target_handle):
+        self.actor_handles.append(target_handle)
+
+    def remove_target(self, target_handle):
+        found_idx = None
+        for i, handle in enumerate(self.actor_handles):
```

Comment:
This looks like `self.actor_handles` should be a dict from `handle._ray_actor_id.hex()` to the handle :)